product_list_price_from_pricelist/models/product_pricelist.py

```diff
@@ -0,0 +1,125 @@
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.tools import config
+
+
+class ProductPricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _update_product_price_from_pricelist(self, specific_pricelist_items=None):
+        self.ensure_one()
+        all_product_ids = set()
+        pricelist_items = (
+            self.item_ids if not specific_pricelist_items else specific_pricelist_items
+        )
```

Comment:
No need to put long names:

```suggestion
    def _update_product_price_from_pricelist(self, pricelist_items=None):
        self.ensure_one()
        all_products = env["product.product"]
        if not pricelist_items:
            pricelist_items = self.item_ids
```